burr/integrations/haystack.py

```diff
@@ -68,47 +142,68 @@ def inputs(self) -> tuple[dict[str, str], dict[str, str]]:
+        """Call the Haystack `Component.run()` method.
 
-        Values come from 3 sources:
-        - bound parameters (from HaystackAction instantiation, or by using `.bind()`)
+        :param state: State object of the application. It contains some input values
+           for ``Component.run()``.
+        :param run_kwargs: User-provided inputs for ``Component.run()``.
+        :return: Dictionary of results with mapping ``{socket_name: value}``.
+
+        Note, values come from 3 sources:
         - state (from previous actions)
-        - run_kwargs (inputs from `Application.run()`)
+        - run_kwargs (inputs from ``Application.run()``)
+        - bound parameters (from ``HaystackAction`` instantiation)
         """
         values = {}
 
         # here, precedence matters. Alternatively, we could unpack all dictionaries at once
         # which would throw an error for key collisions
-        for param, value in self._bound_params.items():
-            values[param] = value
+        for input_socket_name, value in self._bound_params.items():
+            values[input_socket_name] = value
 
-        for param in self.reads:
-            values[param] = state[param]
+        for input_socket_name, state_field_name in self._input_socket_mapping.items():
+            try:
+                values[input_socket_name] = state[state_field_name]
+            except KeyError as e:
+                raise ValueError(f"No value found in state for field: {state_field_name}") from e
 
-        for param, value in run_kwargs.items():
-            values[param] = value
+        for input_socket_name, value in run_kwargs.items():
+            values[input_socket_name] = value
 
         return self._component.run(**values)
 
     def update(self, result: dict, state: State) -> State:
-        """Update the state using the results of `Component.run()`."""
+        """Update the state using the results of ``Component.run()``.
+        The output socket name is mapped to the Burr state field name.
+
+        Values returned by ``Component.run()`` that aren't in ``writes`` are ignored.
+        """
+        # TODO we could want to handle ``.update()`` and ``.append()`` differently
         state_update = {}
-        for socket_name, value in result.items():
-            state_field_name = self._socket_mapping.get(socket_name, socket_name)
-            if state_field_name in self.writes:
-                state_update[state_field_name] = value
 
+        for state_field_name, output_socket_name in self._output_socket_mapping.items():
+            if state_field_name in self.writes:
+                try:
+                    state_update[state_field_name] = result[output_socket_name]
+                except KeyError as e:
+                    raise ValueError(
+                        f"Socket `{output_socket_name}` missing from output of `Component.run()`"
+                    ) from e
+        print(state_update)
```

Comment:
Something odd happened with the review because the current version doesn't have the `print()` and I removed `.bind()` as we've discussed. Also, docstrings are much more complete

the comments appear as `Outdated`